tools/java/org.hl7.fhir.tools.core/src/org/hl7/fhir/tools/implementations/ember/EmberModel.java

```diff
@@ -0,0 +1,359 @@
+                    if (!fieldName.equals("Id")) {
+                        block.ln(getFieldDefinition(fieldName, m.typify("string")));
+                    }
+                } else if (elementType.equals("idref")) {
+                    block.ln(getFieldDefinition(fieldName, m.typify("Reference")));
+                } else if (elementType.equals("Resource")) {
+                    block.ln(getFieldDefinition(fieldName, m.typify("interface{}")));
+                } else if (elementType.equals("Age") || elementType.equals("Count") || elementType.equals("Duration") || elementType.equals("Money")) {
+                    block.ln(getFieldDefinition(fieldName, m.typify("Quantity")));
+                } else {
+                    if (typeRef.isElementReference()) {
+                        elementType = getComponentStructName(elementType);
+                    }
+                    block.ln(getFieldDefinition(fieldName, m.typify(elementType)));
+                }
+            }
+        }
+    }
+
+    private static final class Modifier {
+        private final boolean isSlice;
+
+        public Modifier(ElementDefn el) {
+            this.isSlice = el.getMaxCardinality() == null || el.getMaxCardinality() > 1;
+        }
+
+        public String typify(String type) {
+            type = dasherize(type);
+            String result = type;
+            if (isSlice && ! "string".equals(type) && ! "interface{}".equals(type) && ! "number".equals(type) && !"date".equals(type) && !"boolean".equals(type)) {
+                result = "DS.hasMany('".concat(type).concat("', {embedded: true})");
+            } else if (! "string".equals(type) && ! "interface{}".equals(type) && ! "number".equals(type) && !"date".equals(type) && !"boolean".equals(type)) {
+                result = "DS.belongsTo('".concat(type).concat("', {embedded: true})");
+            } else {
+              result = "DS.attr('" + type + "')";
+            }
+            return result;
+        }
+    }
+
+    private String getFieldName(ElementDefn elementDefinition, TypeRef type) {
+        StringBuilder elementName = new StringBuilder(elementDefinition.getName().replace("[x]", ""));
+        if (elementDefinition.getTypes().size() > 1) {
+            elementName.append(capitalize(type.getName()));
+        }
+        if ("data".equals(elementName.toString())) {
+          return "foo";
+        }
+        return capitalize(elementName.toString());
+    }
+
+    private String getFieldDefinition(String fieldName, String fieldTypeModifier, String fieldType) {
+        String typeName = lowercase(fieldName);
+        fieldName = lowercase(fieldName);
+        return String.format("%s:  %s('%s', {embedded: true})", fieldName, fieldTypeModifier, fieldType, typeName, typeName);
+    }
+
+    private String getFieldDefinition(String fieldName, String fieldType) {
+        String typeName = lowercase(fieldName);
+        return String.format("%s: %s", typeName, fieldType, typeName, typeName);
```

Comment:
There are 2 placeholders in the string format, but 4 arguments are passed in. Assuming you only want the first 2, you can delete the two `typeName` arguments at the end.